lib/truffle/rbconfig-for-mkmf.rb

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+# Split from rbconfig.rb since we need to do expensive tasks like checking the
+# clang and opt versions.
+
+require 'rbconfig'
+
+clang = 'clang'
+opt = 'opt'
+
+extra_cflags = nil
+
+[clang, opt].each do |tool|
+  begin
+    version = `#{tool} --version`
+  rescue Errno::ENOENT
+    raise "#{tool} does not appear to be available - you may need to install LLVM - see doc/user/installing-llvm.md"
+  end
+
+  if version =~ /\bversion (\d+\.\d+\.\d+)/
```

Comment:
It would be nice if there was a built in utility for version checking. In other places I used Gem Version checking.

Reply:
It's 3 lines of Ruby code, could be worse :)
Requiring RubyGems here seems heavy.